fontforge/parsettf.c

```diff
@@ -4716,30 +4716,43 @@ return;
 	    len = getlong(ttf);
 	    /* language = */ getlong(ttf);
 	}
+	uint16 *dtable = NULL;
+	size_t dlen = len;
+	if (dlen >= 1048576) {
+		// One typeface has over 100,000 entries in a table!
+		IError("Table too large; truncated to 1048576 entries.");
+		dlen = 1048576;
+	}
 	if ( enc->is_unicodebmp && (format==8 || format==10 || format==12))
 	    enc = FindOrMakeEncoding("UnicodeFull");
 
 	if ( format==0 ) {
-	    if ( justinuse==git_normal && map!=NULL && map->enccount<256 ) {
-		map->map = realloc(map->map,256*sizeof(int));
-		memset(map->map,-1,(256-map->enccount)*sizeof(int));
-		map->enccount = map->encmax = 256;
+	    dtable = calloc(dlen, sizeof(uint16));
+	    int encmapminsize = dlen;
+	    if (encmapminsize < 256) encmapminsize = 256;
+	    if ( justinuse==git_normal && map!=NULL && map->enccount<encmapminsize ) {
+		map->map = realloc(map->map,encmapminsize*sizeof(int));
+		memset(map->map,-1,(encmapminsize-map->enccount)*sizeof(int));
```

Comment:
Seems weird that it would memset the whole lot and not just the reallocated part, although the old code does something similar

Reply:
That seems wrong now that you mention it. Let me look at it more closely.